lib/socket/clientSocket.hpp

```diff
@@ -9,8 +9,8 @@
 /* Everything we need for socket communication bundled up into a struct */
 typedef struct
 {
-  uint32_t sock;
-  uint32_t nBytesRecv;
+  int sock;
```

Comment:
The items are kept as specifically uint32_t as to be very very careful with the exact number of bytes we're using and how we're using them. Do you have a good reason for this change?

Reply:
The sock member stores the file descriptor returned by the socket() syscall. The manpage is (here)[http://man7.org/linux/man-pages/man2/socket.2.html]. It returns -1 if there's an error, but by assigning it to an unsigned member you lose that information and can't tell if there was an error. There's an if statement in both cpp files checking if (sock < 0), which is always false with an unsigned value.

It could be changed from int to int32_t instead, but I bet int is 32 bits on the pi anyways

Reply:
Hmmm... Unfortunately, testing this branch with the HAB setup, it totally breaks the messaging system. I think you're right about `sock` better off being unsigned, but making `nBytesRecv` a `ssize_t` totally mucks up the socket communication. 

It's also important to note that we want Cubium to be able to run on more than just Raspberry Pi, which is why for any code that interacts with system calls we're very very careful to specify byte numbers. 

I'll happily approve the PR if you'd like to make it `int32_t sock` and `uint32_t nBytesRecv` instead :)